tests/unit/test_timeseries.py

```diff
@@ -146,6 +148,157 @@ def test_timeseries_mul(ts, inplace):
     npt.assert_allclose(ts2.values, [2, 4, 6])
 
 
+@pytest.fixture(scope="function")
+def ts_gtc_per_yr_units():
+    times = np.asarray(
+        [datetime(2000, 1, 1), datetime(2001, 1, 1), datetime(2002, 1, 1),]
+    )
+    return TimeSeries([1, 2, 3], time=times, attrs={"unit": "GtC / yr"})
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_timeseries_add_pint(ts_gtc_per_yr_units, inplace):
+    to_add = 2 * ur(ts_gtc_per_yr_units.meta["unit"])
+    if inplace:
+        ts_gtc_per_yr_units += to_add
+        ts2 = ts_gtc_per_yr_units
+    else:
+        ts2 = ts_gtc_per_yr_units + to_add
+
+    npt.assert_allclose(ts2.values, [3, 4, 5])
+    assert ts2.meta["unit"] == "gigatC / a"
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_timeseries_sub_pint(ts_gtc_per_yr_units, inplace):
+    to_sub = 2 * ur(ts_gtc_per_yr_units.meta["unit"])
+    if inplace:
+        ts_gtc_per_yr_units -= to_sub
+        ts2 = ts_gtc_per_yr_units
+    else:
+        ts2 = ts_gtc_per_yr_units - to_sub
+
+    npt.assert_allclose(ts2.values, [-1, 0, 1])
+    assert ts2.meta["unit"] == "gigatC / a"
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_timeseries_mul_pint(ts_gtc_per_yr_units, inplace):
+    to_mul = 2 * ur("yr")
+    if inplace:
+        ts_gtc_per_yr_units *= to_mul
+        ts2 = ts_gtc_per_yr_units
+    else:
+        ts2 = ts_gtc_per_yr_units * to_mul
+
+    npt.assert_allclose(ts2.values, [2, 4, 6])
+    assert ts2.meta["unit"] == "gigatC"
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_timeseries_div_pint(ts_gtc_per_yr_units, inplace):
+    to_div = 2 * ur("yr**-1")
+    if inplace:
+        ts_gtc_per_yr_units /= to_div
+        ts2 = ts_gtc_per_yr_units
+    else:
+        ts2 = ts_gtc_per_yr_units / to_div
+
+    npt.assert_allclose(ts2.values, [1 / 2, 2 / 2, 3 / 2])
+    assert ts2.meta["unit"] == "gigatC"
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+@pytest.mark.parametrize("op", ["add", "sub"])
+def test_timeseries_add_sub_pint_no_units(ts, inplace, op):
+    scalar = 2 * ur("GtC / yr")
+
+    error_msg = re.escape("Cannot convert from 'dimensionless' to 'gigatC / a'")
+    with pytest.raises(pint.errors.DimensionalityError, match=error_msg):
+        if inplace:
+            if op == "add":
+                ts += scalar
+            elif op == "sub":
+                ts -= scalar
+            else:
+                raise NotImplementedError(op)
+        else:
+            if op == "add":
+                ts + scalar
+            elif op == "sub":
+                ts - scalar
+            else:
+                raise NotImplementedError(op)
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+@pytest.mark.parametrize("op", ["mul"])
+def test_timeseries_mul_pint_no_units(ts, inplace, op):
+    scalar = 2 * ur("GtC / yr")
+    if inplace:
+        if op == "mul":
```

Comment:
Maybe lose the op checks and parameter. Will there be another param which follows the same behaviour?

Reply:
Yep nice done